### PR TITLE
versatiles: init at 0.12.10

### PR DESCRIPTION
**Pull request by wilhelmines**
## VersaTiles

This is a pull request to add VersaTiles to nixpkgs.

The short description from the github repo:
VersaTiles - A toolbox for converting, checking and serving map tiles in various formats.
VersaTiles is a Rust-based project designed for processing and serving tile data efficiently. It supports multiple tile formats and offers various functionalities for handling tile data.

More information can be found on the project's website https://versatiles.org as well as on the GitHub repo: https://github.com/versatiles-org/versatiles-rs

Currently, I'm not included in the development of VersaTiles, but have decided to create the package based on a wish on their contributing page. I plan to be the package maintainer. The lead developer knows about my effort and helped me with fixing related bugs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ve/versatiles/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "versatiles";
+  version = "0.12.9"; # When updating: Replace with current version
+
+  src = fetchFromGitHub {
+    owner = "versatiles-org";
+    repo = "versatiles-rs";
+    rev = "a1896bc1e639ec46bd00f89bffaba9181614515a"; # When updating: Replace with long commit hash of new version
```
Comment (herbetom) [inline pkgs/by-name/ve/versatiles/package.nix:14]:
```suggestion
    rev = "v${version}";
```

I think it's more or less convention to just use the tag here. The hash ensures that noone tampers with the tag.

**Review by herbetom — CHANGES_REQUESTED** [review]
[v0.12.10](https://github.com/versatiles-org/versatiles-rs/releases/tag/v0.12.10) got released in the meantime.

**Comment by herbetom** [comment]
With #350343 and the [`versatiles: adding changelog link`](https://github.com/NixOS/nixpkgs/pull/350343/commits/c45f3ffadf95a1228c058cb4660d979bfc4ab74f) commit in mind: Maybe take a look at the `git rebase --interactive HEAD~3` command (where `HEAD~3` is the number of commits but there are a number of options to specify which commits you want to interact with). This allows you to edit, reword, squash and do other stuff with commits :)

**Comment by herbetom** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350343`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>versatiles</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>versatiles</li>
  </ul>
</details>

File: pkgs/by-name/ve/versatiles/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/ve/versatiles/package.nix:14]: Better
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ve/versatiles/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    description = "VersaTiles - A toolbox for converting, checking and serving map tiles in various formats.";
+    longDescription = ''
+      VersaTiles is a Rust-based project designed for processing and serving tile data efficiently.
+      It supports multiple tile formats and offers various functionalities for handling tile data.
+    '';
+    homepage = "https://versatiles.org/";
+    downloadPage = "https://github.com/versatiles-org/versatiles-rs";
+    changelog = "https://github.com/versatiles-org/versatiles-rs/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ wilhelmines ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/versatiles/package.nix:47]:
```suggestion
  meta = {
    description = "VersaTiles - A toolbox for converting, checking and serving map tiles in various formats.";
    longDescription = ''
      VersaTiles is a Rust-based project designed for processing and serving tile data efficiently.
      It supports multiple tile formats and offers various functionalities for handling tile data.
    '';
    homepage = "https://versatiles.org/";
    downloadPage = "https://github.com/versatiles-org/versatiles-rs";
    changelog = "https://github.com/versatiles-org/versatiles-rs/releases/tag/v${version}";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ wilhelmines ];
```

File: pkgs/by-name/ve/versatiles/package.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    mainProgram = "versatiles";
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+      "x86_64-windows"
+    ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/versatiles/package.nix:55]:
```suggestion
    platforms = with lib.platforms; linux ++ darwin ++ windows;
```

File: pkgs/by-name/ve/versatiles/package.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "--skip tools::convert::tests::test_remote2"
+    "--skip tools::probe::tests::test_remote"
+    "--skip tools::serve::tests::test_remote"
+    "--skip utils::io::data_reader_http"
+    "--skip utils::io::data_reader_http::tests::read_range_git"
+    "--skip utils::io::data_reader_http::tests::read_range_googleapis"
+  ];
+
+  meta = with lib; {
+    description = "VersaTiles - A toolbox for converting, checking and serving map tiles in various formats.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/versatiles/package.nix:38]:
```suggestion
    description = "Toolbox for converting, checking and serving map tiles in various formats.";
```

**Review by Aleksanaa — COMMENTED** [review]
When you have done all the changes, please squash into two commits only: `maintainers: add wilhelmines` and `versatiles: init at 0.12.9`

File: pkgs/by-name/ve/versatiles/package.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    "--skip tools::convert::tests::test_remote2"
+    "--skip tools::probe::tests::test_remote"
+    "--skip tools::serve::tests::test_remote"
+    "--skip utils::io::data_reader_http"
+    "--skip utils::io::data_reader_http::tests::read_range_git"
+    "--skip utils::io::data_reader_http::tests::read_range_googleapis"
+  ];
+
+  meta = {
+    description = "A toolbox for converting, checking and serving map tiles in various formats.";
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/versatiles/package.nix:38]:
```suggestion
    description = "Toolbox for converting, checking and serving map tiles in various formats";
```

**Comment by wilhelmines** [comment]
Hey,
thanks for your responses.
I implemented all of your suggestions.
- The init version is now v0.12.10
- rev refers to the version instead of the commit hash
- The description is shortend
- lib is only used where necessary
- platform syntax is changed
- commits are squashed

If I missed anything, I'm happy to get a response.

**Comment by wilhelmines** [comment]
I made a mistake. How can I remove the other commits that are not part of this PR?

**Comment by wilhelmines** [comment]
I found the solution by myself.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**